src/metalookup/app/api.py

```diff
@@ -92,18 +93,36 @@ async def ping():
 
     # Note: needs to be in global context to be pickleable in order to be transferable to other process.
     def _loop_urls(urls: list[HttpUrl]):
-        async def loop():
-            logging.info(f"Starting cache warmup with {len(urls)} urls")
-            async with ClientSession() as session:
-                for url in urls:
-                    logging.info(f"Warming up {url}")
-                    await session.post(
-                        url=f"http://localhost:{metalookup.lib.settings.API_PORT}/extract?extra=true",
-                        json=Input(url=url).dict(),
-                    )
-            logging.info(f"Finished cache warmup with {len(urls)} urls")
-
-        asyncio.run(loop())
+        # Note:
```

Comment:
Staring at it now this looks like the wrong place for the _loop_urls function - this function is not part of the API currently and just needs to be called from outside. IMHO it would thus make sense to place it elsewhere, to separate it from the API location-wise

Reply:
I moved it to a dedicated file `metalookup/caching/warmup.py`